src/components/MainCarousel.tsx

```diff
@@ -0,0 +1,150 @@
+  width('w-64', 'xs:w-80', 'md:w-104'),
+  display('inline-flex'),
+  overflow('overflow-scroll'),
+  scrollSnap('snap-x', 'snap-mandatory')
+)
+const slide = classnames(
+  width('w-full'),
+  display('flex'),
+  flex('flex-none'),
+  alignItems('items-center'),
+  justifyContent('justify-center'),
+  scrollSnap('snap-start'),
+  textAlign('text-center'),
+  fontSize('text-xl'),
+  textColor('text-white')
+)
+const indicatorContainer = classnames(
+  display('flex'),
+  flexDirection('flex-row'),
+  alignItems('items-center'),
+  justifyContent('justify-center'),
+  gap('gap-1')
+)
+const indicator = (isActive: boolean) =>
+  classnames(
+    width({ 'w-1.5': !isActive, 'w-2': isActive }),
+    height({ 'h-1.5': !isActive, 'h-2': isActive }),
+    backgroundColor({ 'bg-blue-light': !isActive, 'bg-secondary': isActive }),
+    borderRadius('rounded-full'),
+    opacity({ 'opacity-50': !isActive })
+  )
+const indicatorButton = classnames(margin('mx-2'))
+
+const messages = [
+  '🤫 Most used anon app for YC founders',
+  '🦄 #1 to speak anon to unicorn YC founders',
+  '✅ Over 400 verified yc founders, VC-back founders, and VCs',
+  '🌶 #1 app for spicy anon gossip for founders and VCs',
+  '😎 Best anon app for elite founders and VCs in their field',
+]
+const slideDuration = 2500
+
+export default function MainCarousel() {
+  const scrollContainerRef = useRef<HTMLDivElement>(null)
+
+  const [currentSlide, setCurrentSlide] = useState<number>(0)
+  const prevSlide = currentSlide === 0 ? messages.length - 1 : currentSlide - 1
+  const nextSlide = (currentSlide + 1) % messages.length
+
+  const handleNextSlide = useCallback(() => {
+    scrollContainerRef.current?.scrollBy({
+      behavior: 'smooth',
+      left: scrollContainerRef.current?.offsetWidth,
+    })
+  }, [])
+
+  const handlePrevSlide = useCallback(() => {
+    scrollContainerRef.current?.scrollBy({
+      behavior: 'smooth',
+      left: -scrollContainerRef.current?.offsetWidth,
+    })
+  }, [])
+
+  const handleScrollEnd = useCallback(
+    (e: Event) => {
+      if (!(e.target instanceof HTMLElement)) return
+      if (e.target.scrollLeft === 0) setCurrentSlide(prevSlide)
+      if (
+        scrollContainerRef.current?.offsetWidth &&
+        e.target.scrollLeft > scrollContainerRef.current?.offsetWidth
+      )
+        setCurrentSlide(nextSlide)
+    },
+    [prevSlide, nextSlide]
+  )
+
+  useLayoutEffect(() => {
+    scrollContainerRef.current?.scrollTo({
+      left: scrollContainerRef.current?.offsetWidth,
+    })
+  }, [currentSlide])
+
+  useEffect(() => {
+    const interval = setInterval(() => handleNextSlide(), slideDuration)
+    return () => clearInterval(interval)
+  }, [handleNextSlide])
+
+  useEffect(() => {
+    const scrollContainer = scrollContainerRef.current
+    scrollContainer?.addEventListener('scrollend', handleScrollEnd)
+    return () =>
+      scrollContainer?.removeEventListener('scrollend', handleScrollEnd)
+  }, [handleScrollEnd])
+
+  return (
```

Comment:
Let's extract this component into `Carousel` and pass messages to it in this component

Reply:
I think this is a bit premature, but we can definitely do that if we have another version of this in the future!